### ISSUE TITLE
JIT: consider rewrite `movs` logic for struct copies

### ISSUE THREAD

**Issue by markples** [dotnet/runtime#90196]
See discussion in #80086

**Comment by jakobbotsch** [comment]
@EgorBo Did we ever end up fixing this? I recall you had some PRs around this, but that block copies perhaps still aren't handled.

**Comment by EgorBo** [comment]
> [@EgorBo](https://github.com/EgorBo) Did we ever end up fixing this? I recall you had some PRs around this, but that block copies perhaps still aren't handled.

Yep, I only optimized it for inits and only for cases where we can emit SIMD. One thing I remember that `rep movs` is well aligned with write barriers by the registers they use, so if we switch to normal copy instructions, then we might see spills before/after write barriers

**Comment by EgorBo** [comment]
Closed by https://github.com/dotnet/runtime/pull/128542
